fletcher/base.py

```diff
@@ -310,6 +353,158 @@ def base(self):
         """Return base object of the underlying data."""
         return self.data
 
+    def _reduce(self, name, skipna=True, **kwargs):
+        """
+        Return a scalar result of performing the reduction operation.
+
+        Parameters
+        ----------
+        name : str
+            Name of the function, supported values are:
+            { any, all, min, max, sum, mean, median, prod,
+            std, var, sem, kurt, skew }.
+        skipna : bool, default True
+            If True, skip NaN values.
+        **kwargs
+            Additional keyword arguments passed to the reduction function.
+            Currently, `ddof` is the only supported kwarg.
+
+        Returns
+        -------
+        scalar
+
+        Raises
+        ------
+        TypeError : subclass does not define reductions
+        """
+        if name == "any" and pa.types.is_boolean(self.dtype.arrow_dtype):
+            return any_op(self.data, skipna=skipna)
+        elif name == "all" and pa.types.is_boolean(self.dtype.arrow_dtype):
+            return all_op(self.data, skipna=skipna)
+        elif name == "sum" and self.dtype._is_numeric:
+            # TODO: skipna
```

Comment:
maybe we should raise NotImplementedError?

Reply:
This is actually tested and implemented, I removed the TODO and propagated the `skipna` parameter.